pallets/staking/src/lib.rs

```diff
@@ -569,395 +514,428 @@ impl Default for Releases {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
-        /// Mode of era forcing.
-        pub ForceEra get(fn force_era) config(): Forcing;
-
-        /// The percentage of the slash that is distributed to reporters.
-        ///
-        /// The rest of the slashed value is handled by the `Slash`.
-        pub SlashRewardFraction get(fn slash_reward_fraction) config(): Perbill;
+        /// Handler for the unbalanced reduction when slashing a staker.
+        type Slash: OnUnbalanced<NegativeImbalanceOf<Self>>;
 
-        /// The amount of currency given to reporters of a slash event which was
-        /// canceled by extraordinary circumstances (e.g. governance).
-        pub CanceledSlashPayout get(fn canceled_payout) config(): BalanceOf<T>;
+        /// Number of sessions per era.
+        type SessionsPerEra: Get<SessionIndex>;
 
-        /// All unapplied slashes that are queued for later.
-        pub UnappliedSlashes:
-            map hasher(twox_64_concat) EraIndex => Vec<UnappliedSlash<T::AccountId, BalanceOf<T>>>;
+        /// Number of eras that staked funds must remain bonded for.
+        type BondingDuration: Get<EraIndex>;
 
-        /// A mapping from still-bonded eras to the first session index of that era.
+        /// Number of eras that slashes are deferred by, after computation.
         ///
-        /// Must contains information for eras for the range:
-        /// `[active_era - bounding_duration; active_era]`
-        BondedEras: Vec<(EraIndex, SessionIndex)>;
-
-        /// All slashing events on validators, mapped by era to the highest slash proportion
-        /// and slash value of the era.
-        ValidatorSlashInEra:
-            double_map hasher(twox_64_concat) EraIndex, hasher(twox_64_concat) T::AccountId
-            => Option<(Perbill, BalanceOf<T>)>;
-
-        /// Slashing spans for stash accounts.
-        SlashingSpans get(fn slashing_spans): map hasher(twox_64_concat) T::AccountId => Option<slashing::SlashingSpans>;
-
-        /// Records information about the maximum slash of a stash within a slashing span,
-        /// as well as how much reward has been paid out.
-        SpanSlash:
-            map hasher(twox_64_concat) (T::AccountId, slashing::SpanIndex)
-            => slashing::SpanRecord<BalanceOf<T>>;
-
-        /// The earliest era for which we have a pending, unapplied slash.
-        EarliestUnappliedSlash: Option<EraIndex>;
-
-        /// Flag to control the execution of the offchain election. When `Open(_)`, we accept
-        /// solutions to be submitted.
-        pub EraElectionStatus get(fn era_election_status): ElectionStatus<T::BlockNumber>;
-
-        /// True if the current **planned** session is final. Note that this does not take era
-        /// forcing into account.
-        pub IsCurrentSessionFinal get(fn is_current_session_final): bool = false;
```

Comment:
Here the default is given: false, the new version of the code does not give the default value, is there a problem here?

Reply:
because the default bool type value is false. So when geting IsCurrentSessionFinal  at beginning will get false